File: src/Group.py
class Group:
    """Represents one exam group."""

    def __init__(self, group_id: int):
        self.id = group_id
        self.students = []

    def addStudent(self, student):
        if student not in self.students:
            self.students.append(student)

    def containsStudent(self, student) -> bool:
        return student in self.students

    def getGroupSize(self) -> int:
        return len(self.students)

    def __repr__(self) -> str:
        return f"Group(id={self.id}, size={len(self.students)})"

File: Java/src/Group.java
import java.util.ArrayList;

/** Represents one exam group and the students belonging to it. */
public class Group {
    int id;
    ArrayList<Student> students;

    public Group(int ID) {
        this.id = ID;
        this.students = new ArrayList<Student>();
    }

    public int getId() {
        return this.id;
    }

    /** Adds a student to this group if the student is not already present. */
    public void addStudent(Student st) {
        if (!this.students.contains(st)) {
            this.students.add(st);
        }
    }

    public boolean containsStudent(Student st) {
        return this.students.contains(st);
    }

    public int getGroupSize() {
        return this.students.size();
    }
}
